app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_data in people:
+        name = person_data["name"]
+        age = person_data["age"]
+        person = Person(name, age)
+
+        if "wife" in person_data and person_data["wife"]:
+            wife_name = person_data["wife"]
+            if wife_name in Person.people:
+                person.wife = Person.people[wife_name]
+                Person.people[wife_name].husband = person
+
+            elif "husband" in person_data and person_data["husband"]:
+                husband_name = person_data["husband"]
+                if husband_name in Person.people:
+                    person.husband = Person.people[husband_name]
+                    Person.people[husband_name].wife = person
```

Comment:
how will this condition work if the main condition checks `if wife in person_data`. As only 1 parameter can be passed, then this condition will never work